pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -248,13 +252,15 @@ public BrokerResponse handleRequest(JsonNode request, @Nullable SqlNodeAndOption
     if (!hasAccess) {
       _brokerMetrics.addMeteredGlobalValue(BrokerMeter.REQUEST_DROPPED_DUE_TO_ACCESS_ERROR, 1);
       requestContext.setErrorCode(QueryException.ACCESS_DENIED_ERROR_CODE);
+      _brokerQueryEventListener.onQueryCompletion(new BrokerQueryEventInfo(requestContext));
```

Comment:
Looks like we don't increment `requestId` or set it in `requestContext` if there's no access.

Can we move the requestId increment and `requestContext.setRequestId` call before this check?

Reply:
Done.